### ISSUE TITLE
Fixing node position in spring_layout causes a divide by zero exception

### ISSUE THREAD

**Issue by mikaelho** [networkx/networkx#2348]
Sample code:
    import networkx as nx
    
    G=nx.path_graph(4)
    
    fixed_positions = { 1: (0.5, 0.5) }
    fixed_nodes = fixed_positions.keys()
    
    pos=nx.spring_layout(G, pos = fixed_positions, fixed = fixed_nodes)
    
    print(pos)

Output - error suppressed after first run - note all other nodes except the fixed one are nan:
    /private/var/mobile/Containers/Shared/AppGroup/447A26CB-FA57-4E8A-8C34-082F55AD274F/Pythonista3/Documents/site-packages/networkx/drawing/layout.py:342: RuntimeWarning: divide by zero encountered in true_divide
      (k*k/distance**2-A*distance/k))\
    /private/var/mobile/Containers/Shared/AppGroup/447A26CB-FA57-4E8A-8C34-082F55AD274F/Pythonista3/Documents/site-packages/networkx/drawing/layout.py:342: RuntimeWarning: invalid value encountered in true_divide
      (k*k/distance**2-A*distance/k))\
    /private/var/mobile/Containers/Shared/AppGroup/447A26CB-FA57-4E8A-8C34-082F55AD274F/Pythonista3/Documents/site-packages/networkx/drawing/layout.py:342: RuntimeWarning: invalid value encountered in multiply
      (k*k/distance**2-A*distance/k))\
    {0: array([ nan,  nan]), 1: array([ 0.5,  0.5]), 2: array([ nan,  nan]), 3: array([ nan,  nan])}

Environment:
Pythonista 3
Python 3.5
Networkx 1.11

**Comment by mikaelho** [comment]
But this code works as expected (adding the "first" node with an initial position):

    import networkx as nx
    
    G=nx.path_graph(4)
    
    fixed_positions = {0: (0.1, 0.1), 1: (0.5, 0.5) }
    fixed_nodes = [ 1 ]
    
    pos=nx.spring_layout(G, pos = fixed_positions, fixed = fixed_nodes)
    
    print(pos)

**Comment by hagberg** [comment]
This seems to work for me with the 2.0 development version.  Perhaps you could try that?


```python
In [1]: %paste
import networkx as nx
G=nx.path_graph(4)

fixed_positions = { 1: (0.5, 0.5) }
fixed_nodes = fixed_positions.keys()

pos=nx.spring_layout(G, pos = fixed_positions, fixed = fixed_nodes)

print(pos)

## -- End pasted text --
{0: array([ 0.3142939 ,  0.74262896]), 1: array([ 0.5,  0.5]), 2: array([ 0.70349586,  0.23389249]), 3: array([ 0.88815749, -0.00774765])}
```

**Comment by mikaelho** [comment]
Yes, this is fixed in 2.0, tested with 2.0.dev_20170116071359.
